dds/DCPS/transport/framework/TransportReassembly.h

```diff
@@ -96,19 +110,29 @@ class OpenDDS_Dcps_Export TransportReassembly {
 
   struct FragInfo {
     FragInfo()
-      : complete_(false), have_first_(false), range_list_(), total_frags_(0) {}
-    FragInfo(bool hf, const FragRangeList& rl, ACE_UINT32 tf)
-      : complete_(false), have_first_(hf), range_list_(rl), total_frags_(tf) {}
+      : have_first_(false), range_list_(), total_frags_(0) {}
+    FragInfo(bool hf, const FragRangeList& rl, ACE_UINT32 tf, const MonotonicTimePoint& expiration)
+      : have_first_(hf), range_list_(rl), total_frags_(tf), expiration_(expiration) {}
 
-    bool complete_;
     bool have_first_;
     FragRangeList range_list_;
     ACE_UINT32 total_frags_;
+    MonotonicTimePoint expiration_;
   };
 
   typedef OPENDDS_MAP(FragKey, FragInfo) FragInfoMap;
   FragInfoMap fragments_;
 
+  typedef OPENDDS_MULTIMAP(MonotonicTimePoint, FragKey) ExpirationQueue;
+  ExpirationQueue expiration_queue_;
+
+  typedef OPENDDS_MAP_CMP(PublicationId, DisjointSequence, GUID_tKeyLessThan) CompletedMap;
+  CompletedMap completed_;
```

Comment:
Do we need to worry about `completed_` growing without bound?

Reply:
The entries from `completed_` will be removed when no longer associated with that Writer.
The entries in the `DisjointSequence` inside the map will grow in the same way that tracking sequence numbers in the datalink does.